pkg/controller/dnszone/test_helpers.go

```diff
@@ -106,18 +124,36 @@ var (
 		return zone
 	}
 
+	validAzureDNSZoneWithLinkToParent = func() *hivev1.DNSZone {
+		zone := validAzureDNSZone()
+		zone.Spec.LinkToParentDomain = true
+		return zone
+	}
+
 	validDNSZoneWithoutFinalizer = func() *hivev1.DNSZone {
 		zone := validDNSZone()
 		zone.Finalizers = []string{}
 		return zone
 	}
 
+	validAzureDNSZoneWithoutFinalizer = func() *hivev1.DNSZone {
+		zone := validAzureDNSZone()
+		zone.Finalizers = []string{}
+		return zone
+	}
+
 	validDNSZoneWithoutID = func() *hivev1.DNSZone {
 		zone := validDNSZone()
 		zone.Status.AWS = nil
 		return zone
 	}
 
+	validAzureDNSZoneWithoutID = func() *hivev1.DNSZone {
```

Comment:
This is not needed. Use `validAzureDNSZone` everywhere that this function is called.

Reply:
Removed